src/Microsoft.DotNet.XHarness.Apple/Orchestration/RunOrchestrator.cs

```diff
@@ -43,46 +42,41 @@ public interface IRunOrchestrator
     /// We only try to detect the exit code after the app run is finished.
     /// </summary>
     public class RunOrchestrator : BaseOrchestrator, IRunOrchestrator
-    {
-        private readonly IMlaunchProcessManager _processManager;
+    {
+        private readonly IExitCodeDetector _iOSExitCodeDetector;
+        private readonly IExitCodeDetector _macCatalystExitCodeDetector;
         private readonly ILogger _logger;
         private readonly ILogs _logs;
-        private readonly IFileBackedLog _mainLog;
-        private readonly IErrorKnowledgeBase _errorKnowledgeBase;
-        private readonly AppRunner _appRunner;
-
+        private readonly IErrorKnowledgeBase _errorKnowledgeBase;
+        private readonly IAppRunner _appRunner;
+
         public RunOrchestrator(
-            IMlaunchProcessManager processManager,
+            IAppInstaller appInstaller,
+            IAppUninstaller appUninstaller,
+            IAppRunnerFactory appRunnerFactory,
             IDeviceFinder deviceFinder,
+            IExitCodeDetector iOSExitCodeDetector,
+            IExitCodeDetector macCatalystExitCodeDetector,
             ILogger consoleLogger,
             ILogs logs,
             IFileBackedLog mainLog,
             IErrorKnowledgeBase errorKnowledgeBase,
             IDiagnosticsData diagnosticsData,
             IHelpers helpers)
-            : base(processManager, deviceFinder, consoleLogger, logs, mainLog, errorKnowledgeBase, diagnosticsData, helpers)
-        {
-            _processManager = processManager ?? throw new ArgumentNullException(nameof(processManager));
-            _logger = consoleLogger ?? throw new ArgumentNullException(nameof(consoleLogger));
-            _logs = logs ?? throw new ArgumentNullException(nameof(logs));
-            _mainLog = mainLog ?? throw new ArgumentNullException(nameof(mainLog));
+            : base(appInstaller, appUninstaller, deviceFinder, consoleLogger, logs, mainLog, errorKnowledgeBase, diagnosticsData, helpers)
+        {
+            _iOSExitCodeDetector = iOSExitCodeDetector ?? throw new ArgumentNullException(nameof(iOSExitCodeDetector));

```

Comment:
Would it be possible to make the parameters non-nullable to avoid these checks?

Reply:
It is non-nullable, this is a runtime check.

Afaik, it is still best practice to have them (since we also use DI).

Reply:
Do you have some good read about that? I know that nullable reference types are compile time checked only, but a reasonable guidelines about when to use null guards and when not would be definitely useful for me.